sample.py

```diff
@@ -1,95 +1,145 @@
-                define_query = data.read()
-            transaction.query.define(define_query)
-            transaction.commit()
-    print("Opening a Data session to insert data.")
-    with driver.session(DB_NAME, SessionType.DATA) as session:
-        with session.transaction(TransactionType.WRITE) as transaction:
-            with open('iam-data-single-query.tql', 'r') as data:
-                insert_query = data.read()
-            transaction.query.insert(insert_query)
-            transaction.commit()
-        print("Testing the new database.")
-        with session.transaction(TransactionType.READ) as transaction:  # Re-using a session to open a new transaction
-            test_query = "match $u isa user; get $u; count;"
-            response = transaction.query.get_aggregate(test_query)
-            result = response.resolve().as_value().as_long()
-            if result == 3:
-                print("Database setup complete. Test passed.")
+
+def fetch_all_users(driver):
+    print("\nRequest #1: Fetch all users as JSON objects with full names and emails")
+    with driver.session(DB_NAME, SessionType.DATA) as data_session:
+        with data_session.transaction(TransactionType.READ) as read_tx:
+            users = list(read_tx.query.fetch("match $u isa user; fetch $u: full-name, email;"))
+            for i, JSON in enumerate(users):
+                print(f"User #{i + 1}: {JSON}")
+            return users
+
+
+def insert_new_user(driver, name, email):
+    print(f"\nRequest #2: Add a new user with full-name {name} and email {email}")
+    with driver.session(DB_NAME, SessionType.DATA) as data_session:
+        with data_session.transaction(TransactionType.WRITE) as write_tx:
+            response = write_tx.query.insert(
+                f"insert $p isa person, has full-name '{name}', has email '{email}';")
+            counter = 0
+            for res in response:
+                counter += 1
+                name = res.get("_0").as_attribute().get_value()
+                email = res.get("_1").as_attribute().get_value()
+                print("Added new user. Name: " + name + ", E-mail:" + email)
+            write_tx.commit()
+            return counter
+
+
+def get_files_by_user(driver, name, inference=False):
+    if not inference:
+        print(f"\nRequest #3: Find all files that the user {name} has access to view (no inference)")
+        options = TypeDBOptions()
+    else:
+        print(f"\nRequest #4: Find all files that the user {name} has access to view (with inference)")
+        options = TypeDBOptions(infer=True)
+    with driver.session(DB_NAME, SessionType.DATA) as data_session:
+        with data_session.transaction(TransactionType.READ, options) as read_tx:
+            matched_users = list(read_tx.query.get(f"match $u isa user, has full-name '{name}'; get;"))
+            if len(matched_users) > 1:
+                print("WARNING: Found more than one user with tha name.")
+                return []
```

Comment:
Why is this a warning and then returning an empty list instead of continuing?
If it's just a warning, can we continue?
If it's an error condition, we should throw an exception.

Looking at the three conditions, the final else is fired only if there are 0 results, which should mean 0 results for that query as well. Are the cases mixed up?